In [4]:
import time 
import numpy as np

size = 1_000_000
list1 = list(range(size))
arr1 = np.array(list1)

start = time.time()
list_squares = [ x ** 2 for x in list1]
end = time.time()
print("python squaring time is :" ,end-start)

start = time.time()
numpy_squares = arr1 ** 2
end = time.time()
print("Numpy squaring time is :",end-start)

python squaring time is : 0.13163399696350098
Numpy squaring time is : 0.004498720169067383


### Result :-

python squaring time is : 0.13163399696350098

Numpy squaring time is : 0.004498720169067383